File: src/com/github/gtache/editor/DocumentListenerImpl.scala
package com.github.gtache.editor

import com.intellij.openapi.diagnostic.Logger
import com.intellij.openapi.editor.event.{DocumentEvent, DocumentListener}

class DocumentListenerImpl extends DocumentListener {

  private val LOG: Logger = Logger.getInstance(classOf[DocumentListenerImpl])
  private var manager: EditorEventManager = _

  /**
    * Called before the text of the document is changed.
    *
    * @param event the event containing the information about the change.
    */
  override def beforeDocumentChange(event: DocumentEvent): Unit = {
  }


  /**
    * Called after the text of the document has been changed.
    *
    * @param event the event containing the information about the change.
    */
  override def documentChanged(event: DocumentEvent): Unit = {
    if (manager != null) {
      manager.documentChanged(event)
    } else {
      LOG.error("No manager set!")
    }
  }

  def setManager(manager: EditorEventManager): Unit = {
    this.manager = manager
  }
}

File: intellij-lsp/src/main/kotlin/com/github/gtache/lsp/editor/listeners/DocumentListenerImpl.kt
package com.github.gtache.lsp.editor.listeners

import com.intellij.openapi.editor.event.DocumentEvent
import com.intellij.openapi.editor.event.DocumentListener

/**
 * Implementation of a DocumentListener
 */
class DocumentListenerImpl : DocumentListener, AbstractLSPListener() {

    /**
     * Called before the text of the document is changed.
     *
     * @param event the event containing the information about the change.
     */
    override fun beforeDocumentChange(event: DocumentEvent): Unit {
    }


    /**
     * Called after the text of the document has been changed.
     *
     * @param event the event containing the information about the change.
     */
    override fun documentChanged(event: DocumentEvent): Unit {
        if (checkEnabled()) {
            manager?.documentChanged(event)
        }
    }

}